packages/validator/src/services/indices.ts

```diff
@@ -101,15 +122,37 @@ export class IndicesService {
     ApiError.assert(res, "Can not fetch state validators from beacon node");
 
     const newIndices = [];
+
+    const allValidators = new MapDef<SimpleValidatorStatus, number>(() => 0);
+
     for (const validatorState of res.response.data) {
+      // Group all validators by status
+      const status = statusToSimpleStatusMapping(validatorState.status);
+      allValidators.set(status, allValidators.getOrDefault(status) + 1);
+
       const pubkeyHex = toHexString(validatorState.validator.pubkey);
       if (!this.pubkey2index.has(pubkeyHex)) {
-        this.logger.debug("Discovered validator", {pubkey: pubkeyHex, index: validatorState.index});
+        this.logger.debug("Validator exists in beacon chain", {
+          validatorIndex: validatorState.index,
+          pubkey: pubkeyHex,
+        });
         this.pubkey2index.set(pubkeyHex, validatorState.index);
         this.index2pubkey.set(validatorState.index, pubkeyHex);
         newIndices.push(validatorState.index);
       }
     }
+
+    // Filter out validator statuses that are greater than 0
+    const statuses = Object.fromEntries(Array.from(allValidators.entries()).filter((entry) => entry[1] > 0));
+    // The number of validators that are not in the beacon chain
+    const pendingCount = pubkeysHex.length - res.response.data.length;
+    if (pendingCount > 0) {
+      statuses["pending"] += pendingCount;
```

Comment:
This will be NaN if there hasn't been at least one pending entry.
Maybe do this before making `statuses` so you can use the `allValidators.set(s, allValidators.getOrDefault(s) + pendingCount)` pattern